format/volume.go

```diff
@@ -84,13 +84,17 @@ func populateFieldFromBuffer(char rune, buffer []rune, volume *types.ServiceVolu
 		return errors.New("too many colons")
 	}
 	for _, option := range strings.Split(strBuffer, ",") {
-		switch option {
+		key, _, _ := strings.Cut(option, "=")
+
+		switch key {
 		case "ro":
 			volume.ReadOnly = true
 		case "rw":
 			volume.ReadOnly = false
 		case "nocopy":
 			volume.Volume = &types.ServiceVolumeVolume{NoCopy: true}
+		case "subpath":
```

Comment:
The reason is to make it consistent with the Docker CLI where we don't support the short syntax, only the long `--mount`.
We decided to not support it in the short syntax is to avoid growing parsing complexity.

But if compose is fine with that, I'm happy to implement it.

Reply:
right, then maybe better have a generic `default` case rejecting any unsupported attribute

Reply:
The default branch seems to explicitly ignore the unsupported attributes:
https://github.com/compose-spec/compose-go/blob/a1f149ac0ab5fcb5f2954e65dc16300ca4fedfa2/format/volume.go#L98-L102

Also, in this case the option is supported, just not in the short syntax. Returning an "not supported" message could make the user think that it's not supported at all.

Reply:
how does the CLI report attempt to use such an attribute ?

Reply:
CLI errors out when an unknown attribute is used in both long and short syntax.

As for the subpath option specifically, it's not handled differently and will error out just the same as an invalid attribute.
```
$ docker run -v asddd:/v:volume-subpath=sub --rm -it alpine                                                                                                                                                                      
docker: Error response from daemon: invalid mode: volume-subpath=sub.
```

Reply:
right, then I don't think compose should try to be more clever/friendly here _OR_ `subpath` should be made an explicit short-syntax feature